mm/routines/product.py

```diff
@@ -17,11 +17,18 @@ def update_enum_product_attribute(self, sku, att_name, att_value):
         return record
 
     def get_configurable_associated_products(self, store_id=None, is_active=None, is_visible=None):
+        legal_inputs = set([None, True, False])
+        message = "Not a valid input value for '{0}'. Use: {1}"
+
+        assert type(store_id) is int or store_id is None, message.format('store_id', 'None or int')
+        assert is_active in legal_inputs, message.format('is_active', 'None, True or False')
```

Comment:
I know it seems to work fine. That's why most newbies miss it. It's not
correct for mechanical reasons and, apparently, it won't necessarily work
100% of the time. The equals comparison checks in a way that isn't
necessarily correct for one variable operand and one identity operand.
On May 19, 2016 9:41 AM, "Diego Terzano" <notifications@github.com> wrote:

> In mm/routines/product.py
> <https://github.com/CoffeeForThinkers/MagentoModels/pull/4#discussion_r63879701>
> :
>
> > @@ -17,11 +17,18 @@ def update_enum_product_attribute(self, sku, att_name, att_value):
> >          return record
> >
> >      def get_configurable_associated_products(self, store_id=None, is_active=None, is_visible=None):
> > +        legal_inputs = set([None, True, False])
> > +        message = "Not a valid input value for '{0}'. Use: {1}"
> > +
> > +        assert type(store_id) is int or store_id is None, message.format('store_id', 'None or int')
> > +        assert is_active in legal_inputs, message.format('is_active', 'None, True or False')
>
> I works just fine. It only lets pass None, True or False as intended.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/CoffeeForThinkers/MagentoModels/pull/4/files/9aa86cb203d8cf0a259523bfe5481f93b0514543..b90e3dbb8e234665fe945101d8b32cb84d373d25#r63879701>
>